example_scripts/login_system-cache.pwn

```diff
@@ -46,224 +39,286 @@ new Player[MAX_PLAYERS][E_PLAYERS];
-					"Login", "Abort");
-
-			new hashed_pass[129];
-			WP_Hash(hashed_pass, sizeof(hashed_pass), inputtext);
-			
-			if(strcmp(hashed_pass, Player[playerid][Password]) == 0)
-			{
-				//correct password, spawn the player
-				ShowPlayerDialog(playerid, DIALOG_UNUSED, DIALOG_STYLE_MSGBOX, "Login", "You have been successfully logged in.", "Okay", "");
-				Player[playerid][IsLoggedIn] = true;
-				
-			    SetSpawnInfo(playerid, 0, 0, 0.0, 0.0, 0.0, 0.0, 0, 0, 0, 0, 0, 0);
-				SpawnPlayer(playerid);
-			}
-			else
-			{
-			    Player[playerid][LoginAttempts]++;
-			    if(Player[playerid][LoginAttempts] >= 3)
-			    {
-					ShowPlayerDialog(playerid, DIALOG_UNUSED, DIALOG_STYLE_MSGBOX, "Login", CHAT_RED "You have mistyped your password too often (3 times).", "Okay", "");
-			        DelayedKick(playerid);
-				}
-				else
-				    ShowPlayerDialog(playerid, DIALOG_LOGIN, DIALOG_STYLE_PASSWORD, "Login", CHAT_RED "Wrong password!\n" CHAT_WHITE "Please enter your password in the field below:", "Login", "Abort");
-			}
-	    }
+    switch (dialogid)
+    {
+        case DIALOG_UNUSED: return 1; // Useful for dialogs that contain only information and we do nothing depending on whether they responded or not
 	    
-	    case DIALOG_REGISTER:
-	    {
-	        if(!response)
-	            return Kick(playerid);
-	            
-            if(strlen(inputtext) <= 5)
-				return ShowPlayerDialog(playerid, DIALOG_REGISTER, DIALOG_STYLE_PASSWORD, "Registration",
-					CHAT_RED "Your password must be longer than 5 characters!\n" CHAT_WHITE "Please enter your password in the field below:",
-					"Register", "Abort");
-					
-			new query[256];
-			WP_Hash(Player[playerid][Password], 129, inputtext);
-			mysql_format(g_SQL, query, sizeof(query), "INSERT INTO `players` (`username`, `password`) VALUES ('%e', '%s')", Player[playerid][Name], Player[playerid][Password]);
-			mysql_tquery(g_SQL, query, "OnPlayerRegister", "d", playerid);
-	    }
+        case DIALOG_LOGIN:
+        {
+            if (!response) return Kick(playerid);
+
+            new hashed_pass[65];
+            SHA256_PassHash(inputtext, Player[playerid][Salt], hashed_pass, 65);
+
+            if (strcmp(hashed_pass, Player[playerid][Password]) == 0)
+            {
+                //correct password, spawn the player
+                ShowPlayerDialog(playerid, DIALOG_UNUSED, DIALOG_STYLE_MSGBOX, "Login", "You have been successfully logged in.", "Okay", "");
+
+                // sets the specified cache as the active cache so we can retrieve the rest player data
+                cache_set_active(Player[playerid][Cache_ID]);
+                
+                AssignPlayerData(playerid);
+
+                // remove the active cache from memory
+                cache_delete(Player[playerid][Cache_ID]);
+                // the active cache will be unset
+                cache_set_active(MYSQL_INVALID_CACHE);
```

Comment:
`cache_delete` already unsets the active cache if the deleted cache was the active one, so there's no need to explicitly unset it.